modules/comments/comments.php

```diff
@@ -276,7 +276,25 @@ public function comment_form_after() {
 			$params['has_cookie_consent']  = (int) ! empty( $commenter['comment_author_email'] );
 		}
 
-		$signature = Jetpack_Comments::sign_remote_comment_parameters( $params, Jetpack_Options::get_option( 'blog_token' ) );
+		$blog_token = Jetpack_Data::get_access_token();
+		list( $token_key ) = explode( '.', $blog_token->secret, 2 );
+		// Prophylactic check: anything else should never happen.
+		if ( $token_key && $token_key !== $blog_token->secret ) {
+			if ( preg_match( '/^;.\d+;\d+;$/', $token_key, $matches ) ) {
```

Comment:
This regex only matches `;.(only-numbers);(only-numbers);`, but you defined this format earlier as:

`";{$special_id};.{$random_string}"`

These formats seem different.

Reply:
The full Token Secret (the string we think of as "the token") is:

`{$token_key}.{$private}`

For Normal Tokens, both `$token_key` and `$private` are opaque random strings.

For Special Tokens, `$private` is an opaque random string, but `$token_key` looks like:
`";{$special_id};"`

where

`$special_id` looks like:

`.(only-numbers);(only-numbers)`

So `";{$special_id};.{$random_string}"` is the format of the full Special Token, and `";{$special_id};"` matches the Special Token Key format: `;.\d+;\d+;`.

Sounds like I should clean up the docs here :)

Reply:
Thank you for the explanation!